### DISCUSSION TITLE
Can I use HDF5 API to implement custom reference counter?

### DISCUSSION THREAD

**Discussion by tbeu** [HDFGroup/hdf5#4402] [Q&A]
I was wondering if I can use (and hopefully not misuse) the hdf5 framework to implement thread-safe reference counting in C, basically something like std::atomic\<int\>. This way I can mimic a shared pointer in C w/o the need for the typical boilerplate code.

Thank you for your support.

**Comment by derobins** [comment]
Hi. I think the place you want to ask this question is the HDF Forum (https://forum.hdfgroup.org/). We use GitHub discussions to discuss library development.
